src/test/java/steps/CheckboxSteps.java

```diff
@@ -33,31 +33,24 @@ public void user_is_on_checkbox_page(){
 
     }
 
-    @When("User clicks checkbox one {string}")
-    public void user_clicks_checkbox_one(String num){
+    @When("User clicks checkbox {int}")
+    public void user_clicks_checkbox_one(int num){
 
         checkboxPage.clickCheckboxByOrder(checkboxPage.checkboxes, num);
```

Comment:
It is not good practice to get page elements outside of the page object - encapsulation! - Not good practice to provide an element to a method outside of its class because this clickCheckboxByOrder method is only for the Checkbox class. passing elements to method is common practice when there is a generic method for manipulating elements

Reply:
Moved the clickChexboxByOrder method to Checkbox Page and passed list of WebElements directly there. It now accepts only one argument.